pydata_sphinx_theme/__init__.py

```diff
@@ -100,48 +169,58 @@ def docutils_node_to_jinja(list_item, only_pages=False):
     #         <reference> <-- the thing we want
     reference = list_item.children[0].children[0]
     title = reference.astext()
-    url = reference.attributes["refuri"]
+
+    url = reference.attributes.get("refuri", "")
     active = "current" in list_item.attributes["classes"]
 
     # If we've got an anchor link, skip it if we wish
-    if only_pages and '#' in url:
+    if only_pages and "#" in url:
         return None
 
     # Converting the docutils attributes into jinja-friendly objects
     nav = {}
     nav["title"] = title
     nav["url"] = url
     nav["active"] = active
+    nav["type"] = "ref"
 
     # Recursively convert children as well
     # If there are sub-pages for this list_item, there should be two children:
     # a paragraph, and a bullet_list.
     nav["children"] = []
     if len(list_item.children) > 1:
-        # The `.children` of the bullet_list has the nodes of the sub-pages.
-        subpage_list = list_item.children[1].children
-        for sub_page in subpage_list:
-            child_nav = docutils_node_to_jinja(sub_page, only_pages=only_pages)
-            if child_nav is not None:
-                nav["children"].append(child_nav)
+        child_pages = list_item.children[1:]
+        for child in child_pages:
+            # Will either be a caption or another bullet list
+            if isinstance(child, nodes.bullet_list):
+                for subpage in child.children:
+                    this_nav = docutils_node_to_jinja(subpage, only_pages=only_pages)
+                    nav["children"].append(this_nav)
+            else:
+                this_nav = docutils_node_to_jinja(child, only_pages=only_pages)
+                nav["children"].append(this_nav)
     return nav
 
 
 # -----------------------------------------------------------------------------
 
+
 def setup_edit_url(app, pagename, templatename, context, doctree):
     """Add a function that jinja can access for returning the edit URL of a page."""
+
     def get_edit_url():
         """Return a URL for an "edit this page" link."""
         required_values = ["github_user", "github_repo", "github_version"]
         for val in required_values:
             if not context.get(val):
-                raise ExtensionError("Missing required value for `edit this page` button. "
-                                        "Add %s to your `html_context` configuration" % val)
-
-        github_user = context['github_user']
-        github_repo = context['github_repo']
-        github_version = context['github_version']
+                raise ExtensionError(
```

Comment:
I think I may have accidentally black-ified this file in the process 😬 I am not sure how to only revert the black changes, so hoping it's OK